desktop/libs/indexer/src/indexer/indexers/sql.py

```diff
@@ -310,9 +312,45 @@ def create_table_from_local_file(self, source, destination, start_time=-1):
               'columns': ',\n'.join(['  `%(name)s` %(type)s' % col for col in columns]),
             }
 
+    elif editor_type == 'impala':
+      sql = '''CREATE TABLE IF NOT EXISTS %(database)s.%(table_name)s_tmp (
+%(columns)s);
+      ''' % {
+                'database': database,
+                'table_name': table_name,
+                'columns': ',\n'.join(['  `%(name)s` string' % col for col in columns]),
+            }                        #Impala does not implicitly cast between string and numeric or Boolean types.
+
     path = urllib_unquote(source['path'])
 
     if path:                                                     # data insertion
+
+      if editor_type == 'impala':                           # for the boolean col updating csv_col in int value (1,0)
+        bool_list = []
+        for count, col in enumerate(columns):
+          if col['type'] == 'boolean':
+            bool_list.append(count)
+
+        tempfile = NamedTemporaryFile('w', delete=False)
+        with open(BASE_DIR + path, 'r') as local_file, tempfile:
+          reader = csv.reader(local_file)
+          writer = csv.writer(tempfile)
+          if bool_list:
+            for count, row in enumerate(reader):
+              if source['format']['hasHeader'] and count == 0:
+                writer.writerow(row)
+                continue
+              for count in bool_list:
+                if row[count] in ('T', 't', 'true', 'True', 'TRUE', '1'):
+                  row[count] = 1
+                elif row[count] in ('F', 'f', 'false', 'False', 'FALSE', '0'):
+                  row[count] = 0
+                else:
+                  row[count] = None
+              writer.writerow(row)
+            shutil.move(tempfile.name, BASE_DIR + path)
+            tempfile.flush()
+
       with open(BASE_DIR + path, 'r') as local_file:
         reader = csv.reader(local_file)
         list_of_tuples = list(map(tuple, reader))
```

Comment:
Can't we just do it here?

i.e. if Impala we create a new generator on the rows that update the bool columns?